### PR TITLE
Fixes #8519: rudder-dev KeyError: 'master'

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8519

### CODE REVIEW

**Comment by peckpeck** [comment]
There is a little bit of refactoring : I moved get_version to redmine.py
Aside of that the main idea is to add a version_id to issue information, to be able to use it directly when cloning an issue (instead of guessing it)

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/rudder-dev-src [hunk 1]
```diff
@@ -344,22 +345,21 @@ def get_ticket_info(ticket_id, internal=False, must_be_open=True):
...
-    if i in info:
-      new_info[i] = info[i]
-  new_info['parent_issue_id'] = info['id']
+    if i in issue:
+      new_info[i] = issue[i]
+  new_info['parent_issue_id'] = issue['id']
   if bug:
     new_info['tracker_id'] = Config.BUG_TACKER_ID
-    new_info['description'] = "There was a bug in the resolution of #" + str(info['id'])
+    new_info['description'] = "There was a bug in the resolution of #" + issue.name
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev-src]: issue id instead of name?

Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev-src]: or maybe name gives the id of the issue ?

File: scripts/rudder-dev/rudder-dev-src [hunk 2]
```diff
@@ -1053,32 +1064,32 @@ def technique(version, message):
...
         answer = sys.stdin.readline().strip()
         if answer == "":
           answer = next_version
       next_version = answer
+      next_version_id = get_versions()[1][next_version]
   print("Next version will be " + str(next_version))
-  print("Cloning ticket #" + str(info['id']))
-  ticket = clone_ticket(info, next_version, new_title, bug)
-  print(" > ticket #" + str(ticket['id']) + " created " + Config.REDMINE_API_URL + "/issues/" + str(ticket['id']))
+  print("Cloning ticket #" + issue.name)
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev-src]: Same remark on name here

File: scripts/rudder-dev/rudder-dev-src (hunk 1 above)
Comment (peckpeck) [inline scripts/rudder-dev/rudder-dev-src]: name is the id + the "i" for internal issues

**Pull request merged**